src/konnektor/network_tools/misc.py

```diff
@@ -1,37 +1,98 @@
-from typing import Iterable
+from typing import Iterable, Union
 
-from sklearn.base import TransformerMixin, ClusterMixin
-from sklearn.cluster import KMeans
-from scikit_mol.fingerprints import RDKitFingerprintTransformer
+from gufe import Component
+from gufe import LigandNetwork, LigandAtomMapping
 
-from gufe import LigandNetwork, LigandAtomMapping, SmallMoleculeComponent
+from konnektor.network_tools.clustering._abstract_clusterer import \
+    _AbstractClusterer
 
-from konnektor.network_tools.clustering.cluster_components import ComponentsDiversityClustering
 
+def delete_transformation(network: LigandNetwork,
+                          transformation: Union[LigandAtomMapping, tuple[
+                              Component, Component]]) -> LigandNetwork:
+    """
+    Remove the desired edge from the network
 
-def delete_transformation(network :LigandNetwork,
-                          edge :tuple[SmallMoleculeComponent, SmallMoleculeComponent])\
-        ->LigandNetwork:
-    if(isinstance(edge, LigandAtomMapping)):
-        edge = (edge.componentA, edge.componentB)
+    Parameters
+    ----------
+    network: LigandNetwork
+    transformation: :Union[LigandAtomMapping, tuple[Component, Component]]
 
-    f = lambda m: len({m.componentA, m.componentB}.union(edge)) != 2
-    filtered_edges = filter(f, network.edges)
+    Returns
+    -------
+    LigandNetwork
+        returns a copy of the ligand network without the removed edge.
+    """
+    if (isinstance(transformation, LigandAtomMapping)):
+        transformation = (transformation.componentA, transformation.componentB)
+
+    f = lambda m: len({m.componentA, m.componentB}.union(transformation)) != 2
+    filtered_edges = list(filter(f, network.edges))
 
     return LigandNetwork(edges=filtered_edges, nodes=network.nodes)
 
 
-def cluster_compound(compounds: Iterable[SmallMoleculeComponent],
-                     featurize:TransformerMixin = RDKitFingerprintTransformer(),
-                    cluster:ClusterMixin = KMeans(n_clusters=5, n_init="auto")
-                     )->dict[int,
-list[SmallMoleculeComponent]]:
-    
-    clusterer = ComponentsDiversityClustering(featurize = featurize, cluster=cluster)
-    return clusterer.cluster_compounds(compounds)
+def delete_component(network: LigandNetwork,
```

Comment:
maybe this would be a good addition to the LigandNetwork class in gufe

Reply:
on the todo lost, like delete transformation